src/azure-cli/azure/cli/command_modules/botservice/custom.py

```diff
@@ -696,6 +703,13 @@ def update(client, resource_group_name, resource_name, endpoint=None, descriptio
     if app_insights_api_key:
         bot_props.developer_app_insights_api_key = app_insights_api_key
 
+    if cmek_key_vault_url is not None:
+        bot_props.cmek_key_vault_url = cmek_key_vault_url
+        bot_props.is_cmek_enabled = True
+
+    if encryption_off is not None:
```

Comment:
I raised the same thoughts with @munozemilio offline.

The code is saying that the following scenario is valid: a user provides a KeyVault URL but they might want the encryption to be off... Perhaps temporarily?

The module should instead log a warning to inform the user that they provided a KeyVault URL but also turned the encryption off. This would be a better UX for a user doing the above intentional scenario, otherwise the user needs to run to `az bot update` commands twice to mutate their bot resource to the desired state.

Reply:
While the backend code should be fine with the operation, the likelihood that someone tries to do that seems much lower than the likelihood of someone doing that by mistake, so I consider having the error preferable.

Reply:
Changed to throw MutuallyExclusiveArgumentError if both args are passed